src/eins-hwinfo.c

```diff
@@ -103,6 +101,57 @@
 #define COMPUTER_HWINFO_TYPE_STRING "(" RAMINFO_TYPE_STRING \
   ROOTFS_SPACE_TYPE_STRING "@" CPUINFO_ARRAY_TYPE_STRING ")"
 
+/* 24 hours */
+#define RECORD_COMPUTER_HWINFO_INTERVAL_USECONDS (G_USEC_PER_SEC * 60u * 60u * 24u)
+
+/* The path of a file to hold next record time. */
+#define RECORD_TIME_FILE_PATH INSTRUMENTATION_CACHE_DIR "record_time"
+
+static guint64
+get_next_record_time (void)
+{
+  g_autoptr(GKeyFile) kf = g_key_file_new ();
+
+  if (g_key_file_load_from_file (kf,
+                                 RECORD_TIME_FILE_PATH,
+                                 G_KEY_FILE_NONE,
+                                 NULL))
+    return g_key_file_get_uint64 (kf, "hwinfo", "next-record-time", NULL);
+
+  return 0;
+}
+
+/* Get wait time in microseconds */
+static guint64
+get_wait_time_for_next_record (void)
+{
+  guint64 now, record;
+
+  record = get_next_record_time ();
+  now = g_get_real_time ();
```

Comment:
This returns `gint64`, which is signed, not `guint64`, which is not. Why not make get_next_record_time() return a `gint64` and correct the type of these variables?

It is true that in practice these values are always positive. No I don't know why it returns a signed value. :)

This function can still return `guint64`. We know that both branches below return a non-negative number.

Reply:
The datatype bothers me and makes me thinking these days.  The original idea was `time_t`, which is the best one.  However, `g_get_real_time()` return a value with datatype `gint64`.  So, I think where is the best point to unify the datatype as well.